src/reductiveContext.re

```diff
@@ -8,35 +8,50 @@ module type Config = {
 
 module Make = (Config: Config) => {
   open Subscription;
-  let storeContext = React.createContext(Config.store);
+  let storeContext = React.createContext(None);
 
   module ContextProvider = {
     let make = React.Context.provider(storeContext);
-    let makeProps = (~value, ~children, ()) => {
+    let makeProps =
+        (
+          ~value: option(Reductive.Store.t(Config.action, Config.state)),
+          ~children,
+          (),
+        ) => {
       "value": value,
       "children": children,
     };
   };
 
   module Provider = {
     [@react.component]
-    let make = (~children) => {
-      <ContextProvider value=Config.store> children </ContextProvider>;
+    let make = (~children, ~store) => {
+      <ContextProvider value={Some(store)}> children </ContextProvider>;
     };
   };
 
-  let useSelector = selector => {
+  let useStore = () => {
     let storeFromContext = React.useContext(storeContext);
+    switch (storeFromContext) {
+    | None =>
+      failwith(
+        "could not find reductive context value; please ensure the component is wrapped in a <Provider>",
```

Comment:
This is technically still a breaking change even with my recommendation above. Happy to have this addition and bump w/ major.

Super nit: upper case `Could` please.